### PR TITLE
build: Change base Docker image from Alpine to Debian

### PR DESCRIPTION
**Pull request by aleksraiden**
Change base Docker image from alpine 3.16 (as an too old system) into Debian - use bookworm-slim image. (see our issue too https://github.com/apache/kvrocks/issues/2344)

As we can't upgrade alpine, for continuous use latest rocksdb release and fix some mysterious (for me, see https://github.com/apache/kvrocks/pull/2327) bugs with jemalloc in alpine, I propagate an new Docker, based on latest Debian slim. 

Basically, nothing to change from original, just change an image and use apt instead of apk. Added only an apt clean command for cleanup system after upgrade. 

Result image has an ~ 120Mb size instead of ~40 Mb in alpine.

### CODE REVIEW

**Comment by aleksraiden** [comment]
Thanks to @thbley about the some fix and optimizing from https://github.com/apache/kvrocks/issues/2344

**Comment by aleksraiden** [comment]
Result image has an ~ 120Mb size instead of ~40 Mb in alpine. I think that its OK.

**Comment by git-hulk** [comment]
@aleksraiden Thanks for taking care of this.

**Comment by aleksraiden** [comment]
Some troubles in healthcheck on docker, try to fix today or tomorrow

**Comment by PragmaTwice** [comment]
> As we can't upgrade alpine

Hmm is there any issue for us about upgrading alpine?

**Comment by PragmaTwice** [comment]
And also, do you have any suggestion to select between debian and ubuntu?

**Comment by thomas-0816** [comment]
> And also, do you have any suggestion to select between debian and ubuntu?

To me both are quite good and work great in production for many companies.
Ubuntu lts versions have longer support lifetime, more recent versions of packages (e.g. gcc, openssl), Debian more stable, less recent versions. Key for alpine is size and security, but no glibc.

for package version comparison:
https://packages.ubuntu.com/
https://www.debian.org/distrib/packages
https://pkgs.alpinelinux.org/packages

**Comment by aleksraiden** [comment]
Yeah, sorry, I have not any real experience with ubuntu, but over 10 y use Debian on all our servers, so use it as a matured server-side os.

In other hands, the idea of multiple images (like a community or non-official, in separate directory etc) with other base images my Ok and I can support this feature. 

In my mind, ideally - alpine 3.20+ but we cant switch due to some dependency. So - Debian looks like good solution at now.

**Comment by aleksraiden** [comment]
> > As we can't upgrade alpine
> 
> Hmm is there any issue for us about upgrading alpine?

With @torwig we have a lot of internal tests and a core point is libexecinfo dependency

**Review by git-hulk — APPROVED** [review]

**Review by PragmaTwice — APPROVED** [review]
